applicationconfig/client/src/main/java/com/azure/applicationconfig/models/ConfigurationSetting.java

```diff
@@ -194,6 +185,16 @@ public boolean isLocked() {
         return locked;
     }
 
+    /**
+     * Sets the locked status for this configuration setting. This value is not used in service requests.
+     * @param locked The lock status the configuration setting is being set.
+     * @return The updated ConfigurationSetting object.
+     */
+    public ConfigurationSetting isLocked(boolean locked) {
+        this.locked = locked;
+        return this;
+    }
```

Comment:
Didn't we decide to not expose the locking functionality?

Reply:
Correct, the locking functionality is disabled but that was done with removing the client's method that performed locking and unlocking.

The locked property in the ConfigurationSetting doesn't determine the locked status of the configuration. I did this to allow the equals overload to be used in the unit tests as some properties returned by the service were getter only in the model.

Reply:
Removed the two setter methods I added to ConfigurationSetting in favor of using reflections setting those values for the unit tests.

Reply:
if your tests are in the same package name as your actual code, you should instead just use package-private methods rather than reflection. Also, a naming nit: `isLocked(boolean)` is a really bad naming pattern - it is a setter that looks like a getter. It should have been called either `locked(boolean)` or `setLocked(boolean)`, depending on context.

Reply:
They're in different packages unfortunately, so the package-private implementation won't work.

I agree that isLocked should get renamed to locked to be consistent with the other getters and setters.